src/api/telemetry/TelemetryCollection.js

```diff
@@ -0,0 +1,358 @@
+     * data could be added OR removed here we update the current
+     * bounded telemetry and emit the results
+     *
+     * @param  {TimeConductorBounds} bounds The newly updated bounds
+     * @param  {boolean} [tick] `true` if the bounds update was due to
+     * a "tick" event (ie. was an automatic update), false otherwise.
+     */
+    _bounds(bounds, isTick) {
+        let startChanged = this.lastBounds.start !== bounds.start;
+        let endChanged = this.lastBounds.end !== bounds.end;
+
+        this.lastBounds = bounds;
+
+        this.emit('bounds', ...arguments);
+
+        if (isTick) {
+            // need to check futureBuffer and need to check
+            // if anything has fallen out of bounds
+            let startIndex = 0;
+            let endIndex = 0;
+
+            let discarded = [];
+            let added = [];
+            let testDatum = {};
+
+            if (startChanged) {
+                testDatum[this.timeKey] = bounds.start;
+                // Calculate the new index of the first item within the bounds
+                startIndex = _.sortedIndexBy(this.boundedTelemetry, testDatum, datum => this.parseTime(datum));
+                discarded = this.boundedTelemetry.splice(0, startIndex);
+            }
+
+            if (endChanged) {
+                testDatum[this.timeKey] = bounds.end;
+                // Calculate the new index of the last item in bounds
+                endIndex = _.sortedLastIndexBy(this.futureBuffer, testDatum, datum => this.parseTime(datum));
+                added = this.futureBuffer.splice(0, endIndex);
+                this.boundedTelemetry = [...this.boundedTelemetry, ...added];
+            }
+
+            if (discarded.length > 0) {
+                this.emit('remove', discarded);
+            }
+
+            if (added.length > 0) {
+                this.emit('add', added);
+            }
+
+        } else {
+            // user bounds change, reset
+            this._reset();
+        }
+
+    }
+
+    /**
+     * whenever the time system is updated need to update related values in
+     * the Telemetry Collection and reset the telemetry collection
+     *
+     * @param  {TimeSystem} timeSystem - the value of the currently applied
+     * Time System
+     */
+    _timeSystem(timeSystem) {
+        this.timeKey = timeSystem.key;
+        let metadataValue = this.metadata.value(this.timeKey) || { format: this.timeKey };
+        let valueFormatter = this.openmct.telemetry.getValueFormatter(metadataValue);
+
+        this.parseTime = (datum) => {
+            return valueFormatter.parse(datum);
+        };
+
+        this.emit('timesystem', timeSystem);
+        this._reset();
+    }
+
+    /**
+     * Reset the telemetry data of the collection, and re-request
+     * historical telemetry
+     *
+     * @param  {boolean} skipRequest - skip requesting history, default false
+     *
+     * @todo handle subscriptions more granually
+     */
+    _reset(skipRequest = false) {
```

Comment:
Rather than passing a flag, we could just call them separately. `_reset` is for clearing the collection (maybe it should be called `_clear`?) and `_requestHistoricalTelemetry` is for requesting historical telemetry,

Reply:
Yeah I've replaced this totally, it was for that unused updateOptions function, so it's kinda obsolete now. I've renamed to  _clear, I left the requestHistorical in there since it's used in each call to _clear. Let me know if you think I should just pull it out.